src/main/java/io/quarkus/registry/app/DatabaseRegistryClient.java

```diff
@@ -173,14 +187,48 @@ private PlatformCatalog toPlatformCatalog(List<PlatformRelease> platformReleases
         return catalog.build();
     }
 
+    private List<ExtensionOrigin> toExtensionOrigins(ExtensionRelease extensionRelease) {
+        final List<ExtensionOrigin> extensionOrigins;
+
+        if (extensionRelease.platforms.isEmpty()) {
+            String id = getNonPlatformCoords(extensionRelease.quarkusCoreVersion);
+            extensionOrigins = List.of(ExtensionCatalog.builder().setId(id)
+                    .setQuarkusCoreVersion(extensionRelease.quarkusCoreVersion)
+                    .setPlatform(false)
+                    .build());
+        } else {
+            extensionOrigins = extensionRelease.platforms.stream()
+                    .map((platform) -> (ExtensionOrigin) ExtensionCatalog.builder()
+                            .setId(platform.platformRelease.platformStream.streamKey)
+                            .setBom(ArtifactCoords.pom(IO_QUARKUS_PLATFORM, QUARKUS_BOM, platform.platformRelease.version))
```

Comment:
This will need adjustment for sure. Although I am not sure exactly what to suggest here :)
1. I am not sure what `platform.platformRelease.platformStream.streamKey` returns but it should be `<platform-key>:<member-bom-artifactId>-quarkus-platform-descriptor:<quarkus-version>:json:<quarkus-version>`
2. we can't always assume the BOM groupId will be `io.quarkus.platform`, it should be `<platform-key>` or something that's used in the registry for that, perhaps there is a config option for it, because there is also the RH version of the registry
3. we can't always assume the BOM artifactId will be `quarkus-bom`, it's `<member-bom-artifactId>`, e.g. `quarkus-camel-bom`, quarkus-kogito-bom`, etc.

Unless I missed it George adding it, I think the current impl of the registry doesn't actually include any platform extensions besides the core and non-platform ones, which is a big issue here and it affects #92 as well.

Reply:
BTW, there are utility methods in `io.quarkus.registry.util.PlatformArtifacts` to help converting BOM coordinates to the platform catalog coords and back. A catalog id so far has been the `catalogCoords.toString()`.

Reply:
> Unless I missed it George adding it, I think the current impl of the registry doesn't actually include any platform extensions besides the core and non-platform ones, which is a big issue here and it affects #92 as well.

This feature is tracked in https://github.com/quarkusio/registry.quarkus.io/issues/91

Reply:
>     1. I am not sure what `platform.platformRelease.platformStream.streamKey` returns but it should be `<platform-key>:<member-bom-artifactId>-quarkus-platform-descriptor:<quarkus-version>:json:<quarkus-version>`

I think the streamKey is much less complete than that, more like "2.6". :)
The reason I thought it was an acceptable ID was I traced through the code and it seems to be a primary key/entity id in the database. But I can see that I was muddling "entity id" and "maven id" and those two might have have very different requirements. I'll see if I have enough information to construct the more informative version.

Reply:
> 2\. we can't always assume the BOM groupId will be `io.quarkus.platform`, it should be `<platform-key>` or something that's used in the registry for that, perhaps there is a config option for it, because there is also the RH version of the registry

I see your point. I did of course copy and paste that from elsewhere in the code. :) I wondered if that would also need correcting, but looking at it, I think the original source is ok. Just to confirm, for a non-platform extension, is `io.quarkus.platform` always on ok value?

Reply:
The BOM groupId, such as `io.quarkus.platform`, should probably be coming from some registry config. If there isn't such config option, we should introduce it. If one day we introduce a non-platform extensions in `quarkus.registry.redhat.com`, it will use `com.redhat.quarkus.platform` as the groupId, for example.

Reply:
I think I have improved it slightly, from @aloubyansky's "This will need adjustment for sure",  but it's still not perfect.
Well, or even mergeable. 

I can't find a way to get from a `PlatformExtension` to an `ExtensionCatalog`/`ExtensionOrigin`. All I can get is a list of all the member boms in the catalog, but I don't know which one the extension was listed in. I've looked through #92 for inspiration, but didn't spot anything that would obviously help me.

I see the following options: 

1. Live with my hardcoding of the `<member-bom-artifactId>` to `unknown`. Since I don't know the member bom, it's not wrong. :) 
2. I notice @gastaldi 's #92, https://github.com/quarkusio/registry.quarkus.io/pull/92/files#diff-d49f8461f49c6c50a2e13ff42dc0df5d8be9fdfd271d87181d4c4b85db18376e , seems to be hardcoding the bom artifact id to `quarkus-bom` at one point. Especially if the current impl of the registry only includes core and non-platform extensions, that hardcoding would be quite safe (for the moment). 
1. I don't actually need to know the `ExtensionOrigin`, just the platform. So I could update the API to add a new method alongside `getOrigins`, something like `getPlatformStreams`.
3.  I could do the work to get the origins, even though I don't think it's in the database. I think if I queried maven for the listing of every member bom in a platform release, I could work backwards to see if the extension I care about is included. It seems a lot like hard work, though. 
4. For the moment, don't include any origin in what I pass back. That would make the new API less useful than it could be, but definitely much more useful than no API at all, which is what we currently have. And if we don't include information, it can't be wrong. :) 

What do you think, @gastaldi and @aloubyansky?

Reply:
@holly-cummins I think for your PR hardcoding `quarkus-bom` would be ok. We need a proper refactoring of the model to get this right which should be a separate change.

Reply:
Ok, thanks, @aloubyansky! I'll get things tidied up and squash for a final review.

Reply:
Thanks for the feedback, @gastaldi and @aloubyansky. I've got a squashed version pushed now.